packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.ts

```diff
@@ -69,6 +71,8 @@ export const useFluentProvider_unstable = (
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const renderer = useRenderer_unstable();
+  const { styleTagId, rule } = useFluentProviderThemeStyleTag({ theme: mergedTheme, targetDocument });
```

Comment:
We invoke `useRenderer_unstable` twice now 😨 Let's do better and pass them to `useFluentProviderThemeStyleTag()`

```suggestion
  const renderer = useRenderer_unstable();
  const { styleTagId, rule } = useFluentProviderThemeStyleTag({ theme: mergedTheme, targetDocument });
```

Reply:
We will be also simplify a test in `useFluentProviderThemeStyleTag.test.tsx`

```js
  it('should handle "styleAttributes"', () => {
    const { result } = renderHook(() =>
      useFluentProviderThemeStyleTag({
        theme: defaultTheme,
        targetDocument: document,
        styleAttributes: { nonce: 'random' },
      }),
    );
    const tag = document.getElementById(result.current.styleTagId) as HTMLStyleElement;

    expect(tag.getAttribute('id')).toBe('fui-FluentProvider1');
    expect(tag.getAttribute('nonce')).toBe('random');
  });
```